src/Nethermind/Nethermind.Api/Extensions/IPlugin.cs

```diff
@@ -17,16 +17,21 @@
 
 using System;
 using System.Threading.Tasks;
-using Nethermind.Int256;
 
-namespace Nethermind.GasService
+namespace Nethermind.Api.Extensions
 {
-    public class GasPriceOracle : IGasPriceOracle
+    public interface INethermindPlugin : IDisposable
     {
-        public Task<UInt256> GetGasPrice(ulong txGasLimit, GasPriceType priceType)
-        {
-            var provider = provide.NChain.InitNChain("token");
-            throw new NotImplementedException();
-        }
+        string Name { get; }
+        
+        string Description { get; }
+        
+        string Author { get; }
+        
+        Task Init(INethermindApi api);
+        
+        Task InitNetworkProtocol();
+        
+        Task InitRpcModules();
```

Comment:
ValueTask maybe? as its quite optional.

Reply:
just a few calls here on the app startup